app/src/main/java/com/example/android/uamp/fragments/SettingsScreen.kt

```diff
@@ -0,0 +1,209 @@
+            ) {
+                Text(
+                    text = "Enable Spatial Audio", modifier = Modifier
+                        .weight(2f)
+                )
+                Switch(
+                    checked = mCheckedValue.value,
+                    onCheckedChange = {
+                        mCheckedValue.value = it
+                        toggleSpatialAudio(mainActivityViewModel, mCheckedValue.value)
+                    },
+                    colors = SwitchDefaults.colors(
+                        checkedTrackColor = colorResource(id = R.color.colorPrimaryDark),
+                        checkedThumbColor = Color.White,
+                        uncheckedTrackColor = Color.Gray
+                    ),
+                )
+            }
+            SpatialAudioOutput(audioManager = audioManager!!)
+        }
+    }
+}
+
+/**
+ * SpatialAudioOutput initializes a spatializer through which the current spatial audio characteristics
+ * of the current media item being played are displayed.
+ *
+ * @param audioManager to declare spatializer
+ */
+@Composable
+fun SpatialAudioOutput(audioManager: AudioManager) {
+
+    val spatializer = audioManager.spatializer
+
+    val attributes = AudioAttributes.Builder().setUsage(AudioAttributes.USAGE_MEDIA).setContentType(
+        C.AUDIO_CONTENT_TYPE_UNKNOWN
+    ).setAllowedCapturePolicy(C.ALLOW_CAPTURE_BY_ALL).build()
+    val audioFormat = AudioFormat.Builder()
+        .setEncoding(AudioFormat.ENCODING_PCM_16BIT)
+        .setChannelMask(AudioFormat.CHANNEL_OUT_5POINT1)
+        .build()
+
+    val canBeSpatialized = spatializer.canBeSpatialized(attributes, audioFormat)
+    val getImmersiveAudioLevel = spatializer.immersiveAudioLevel
+    val isEnabled = spatializer.isEnabled
+    val isAvailable = spatializer.isAvailable
+
+    // Introduced in API 33, be sure to use a compatible device
+    val isHeadTrackerAvailable = spatializer.isHeadTrackerAvailable
+
+    Column(verticalArrangement = Arrangement.SpaceBetween) {
+        Text(
+            text = "Spatial Audio Output:"
+        )
+        Surface() {
+            Column(verticalArrangement = Arrangement.SpaceBetween) {
+                Row() {
+                    Text(
+                        text = "canBeSpatialized: ",
+                        modifier = Modifier.weight(1f),
+                        fontStyle = FontStyle.Italic,
+                        style = MaterialTheme.typography.h6
+                    )
+                    Text(text = canBeSpatialized.toString())
+                }
+                Row() {
+                    Text(
+                        text = "getImmersiveAudioLevel: ",
+                        modifier = Modifier.weight(1f),
+                        fontStyle = FontStyle.Italic,
+                        style = MaterialTheme.typography.h6
+                    )
+                    Text(text = getImmersiveAudioLevel.toString())
```

Comment:
Instead of displaying the int representation of this value, can we have it display a more helpful string? For example, "Multichannel" for 1, "None" for 0, and "Other" for -1.
https://developer.android.com/reference/android/media/Spatializer#SPATIALIZER_IMMERSIVE_LEVEL_MULTICHANNEL